enhancements/microshift/rhoai-model-serving-on-microshift.md

```diff
@@ -0,0 +1,339 @@
+  to minimize to make the solution suitable for edge devices.
+  - Also done in the configmap.
+- Package the manifests as an RPM `microshift-kserve` for usage.
+
+Provide users with ServingRuntime definitions derived from RHOAI, so they
+are not forced to use upstream manifests.
+Decision on how to do this is pending. See open questions.
+
+### Rebase procedure in detail
+
+1. `rebase_rhoai.sh` script shall take an argument pointing to the RHOAI Operator Bundle Image.
+1. The script will:
+   1. Extract ClusterServiceVersion (CSV) of the RHOAI Operator.
+   1. Obtain RHOAI Operator image reference from the CSV.
+   1. Extract /opt/manifests from the RHOAI Operator image.
+   1. Copy contents of /opt/manifests/kserve to MicroShift repository.
+   1. Copy contents of /opt/manifests/odh-model-controller/runtimes to MicroShift repository.
+   1. Prepare top-level kustomization.yaml with MicroShift-specific tweaks.
+      1. Create kserve settings ConfigMap to configure MicroShift-specific tweaks.
+      1. Use contents of different files from RHOAI Operator's /opt/manifests
+         to get downstream-rebuilt image references and apply them over the
+         existing manifests.
+
+
+### Workflow Description
+
+**User** is a human administrating and using MicroShift cluster/device.
+
+(RPM vs ostree vs bootc is skipped because it doesn't differ from any other MicroShift's RPM).
+
+1. User installs `microshift-kserve` RPM and restarts MicroShift service.
+1. Kserve manifest are deployed.
+1. User configures the hardware, the OS, and additional Kubernetes components to
+   make use of their accelerators.
+1. ServingRuntimes are delivered with the kserve RPM or deployed by the user.
+1. User creates InferenceService CR which references ServingRuntime of their choice
+   and reference to the model.
+1. Kserve creates Deployment, Ingress, and other.
+1. Resources from previous step become ready and user can make HTTP/GRPC calls
+   to the model server.
+
+### API Extensions
+
+`microshift-kserve` RPM will bring following CRDs, however they're not becoming
+part of the core MicroShift deployment:
+- InferenceServices
+- TrainedModels
+- ServingRuntimes
+- InferenceGraphs
+- ClusterStorageContainers
+- ClusterLocalModels
+- LocalModelNodeGroups
+
+Contents of these CRDs can be viewed at https://github.com/red-hat-data-services/kserve/tree/master/config/crd/full.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+Enhancement is MicroShift specific.
+
+#### Standalone Clusters
+
+Enhancement is MicroShift specific.
+
+#### Single-node Deployments or MicroShift
+
+Enhancement is MicroShift specific.
+
+### Implementation Details/Notes/Constraints
+
+See "Proposal"
+
+### Risks and Mitigations
+
+At the time of writing this document, kserve's Raw Deployment mode is not fully
+supported by the RHOAI. For for reason, this feature will start as Tech Preview
+and only advance to GA when RHOAI starts supporting Raw Deployment mode.
```

Comment:
This is going to change in like 1 release so I think we can already assume here to have it fully supported.